src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -472,7 +514,28 @@ private void Initialize()
 
                 Interop.close(fileDescriptor);
                 _registerViewPointer = (RegisterView*)mapPointer;
-            }
+
+                // Detect whether we're running on a Raspberry Pi 4
+                IsPi4 = false;
+                try
+                {
+					if (File.Exists(ModelFilePath))
+					{
+						string model = File.ReadAllText(ModelFilePath, Text.Encoding.ASCII);
+						if (model.Contains("Raspberry Pi 4"))
+						{
+							IsPi4 = true;
+						}
+					}
+                }
+                catch (Exception x)
+				{
+                    // This should not normally fail, but we currently don't know how this behaves on different operating systems. Therefore, we ignore
+					// any exceptions in release and just continue as Pi3 if something fails. 
+                    // If in debug mode, we might want to check what happened here (i.e unsupported OS, incorrect permissions)
+					System.Diagnostics.Debug.Fail($"Unexpected exception: {x}");
```

Comment:
`using System.Diagnostics`

Reply:
@krwq I just saw that `Debug.Fail` has no effect, since DEBUG is not defined. Do we need to add something like 
```
<PropertyGroup Condition="$(Configuration.EndsWith('Debug'))=='true'">
    <DefineConstants>DEBUG;TRACE</DefineConstants>
</PropertyGroup>
```
Or should this be provided by the build scripts?

Reply:
`dotnet build -c Debug` should define it I believe. Create an issue if it still doesn't work, let's refrain from these kind of changes here

Reply:
Created #881. Something is fishy there.